File: worksheets/dp-four/scala/src/proxy/remoteproxy/ReportGeneratorImpl.scala
package proxy.remoteproxy

import java.rmi.Naming
import java.rmi.RemoteException
import java.rmi.server.UnicastRemoteObject
import java.util.Date

object ReportGeneratorImpl extends App {
  try {
    val reportGenerator: ReportGenerator = new ReportGeneratorImpl()
    Naming.rebind("PizzaCoRemoteGenerator", reportGenerator)
  } catch {
    case e: Exception => e.printStackTrace()
  }
}

class ReportGeneratorImpl protected ()
  extends UnicastRemoteObject
  with ReportGenerator {

  override def generateDailyReport(): String = ???
}

File: worksheets/dp-four/kotlin/src/proxy/remoteproxy/ReportGeneratorImpl.kt
package proxy.remoteproxy

import java.rmi.Naming
import java.rmi.server.UnicastRemoteObject

fun main(args: Array<String>) {
    try {
        val reportGenerator: ReportGenerator = ReportGeneratorImpl()
        Naming.rebind("PizzaCoRemoteGenerator", reportGenerator)
    } catch (e: Exception) {
        e.printStackTrace()
    }
}

class ReportGeneratorImpl()
    : UnicastRemoteObject(), ReportGenerator {

    override fun generateDailyReport(): String {
        TODO()
    }
}
